explainaboard/loaders/kg_link_tail_prediction.py

```diff
@@ -29,7 +30,12 @@ def load(self) -> Iterable[dict]:
         :return: class object
         """
         data: list[dict] = []
-        raw_data = self.file_loaders[self._file_type].load_raw(self._data, self._source)
+
+        # TODO(odashi): Avoid potential bug: load_raw returns Iterable[Any] which is not a dict.
+        raw_data: dict[str, dict[str, str]] = self.file_loaders[  # type: ignore
+            unwrap(self._file_type)
+        ].load_raw(self._data, self._source)
```

Comment:
@neubig please give me details about this.

Reply:
KgLinkTailPrediction is a special case I didn't think of. The system outputs are (key, value) pairs instead of a list of data points. `JSONFileLoader` uses `json.loads` which returns an any so I guess we can't guarantee it's an iterable anyways so maybe we should change the function signature of `load_raw()` to return any? What do you think?

Reply:
Yeah, I think `load_raw` loads raw data, so we shouldn't be over-prescriptive of its format.

Also, maybe we should somehow indicate that `load_raw` shouldn't be used outside of `explainaboard.loaders`?

Reply:
Sounds good. Feel free to either do this yourself in this PR or assign an issue to me for completion after this PR is merged.

Reply:
Keeps the todo comment and resolves this conversation.